src/utils.js

```diff
@@ -8,8 +8,16 @@ import createHash from 'murmurhash-js/murmurhash3_gc'
  * @return {String}
  */
 export function generateClassName(str, rule) {
+  const {name, options: {sheet}} = rule
   const hash = createHash(str)
-  return rule.name ? `${rule.name}-${hash}` : hash
+
+  let className = name ? `${name}-${hash}` : hash
+
+  if (sheet && sheet.options.meta) {
+    className = `${sheet.options.meta}_${className}`
```

Comment:
@kof Let's do that.

I'm still going to drop a custom method in for `jss-theme-reactor` so that I can output pretty selectors for debugging with the component/sheet name visible. But in terms of fixing the issue in `jss` core, this is a great solution 👍